frontend/src/screens/wallet/Swaps.tsx

```diff
@@ -0,0 +1,499 @@
+        </p>
+      </div>
+      <Label>Swap To</Label>
+      <RadioGroup
+        defaultValue="normal"
+        value={isInternalSwap ? "internal" : "external"}
+        onValueChange={() => {
+          setDestination("");
+          setInternalSwap(!isInternalSwap);
+        }}
+        className="flex gap-4 flex-row"
+      >
+        <div className="flex items-start space-x-2 mb-2">
+          <RadioGroupItem value="internal" id="internal" className="shrink-0" />
+          <Label
+            htmlFor="internal"
+            className="text-primary font-medium cursor-pointer"
+          >
+            Spending balance
+          </Label>
+        </div>
+        <div className="flex items-start space-x-2">
+          <RadioGroupItem value="external" id="external" className="shrink-0" />
+          <Label
+            htmlFor="external"
+            className="text-primary font-medium cursor-pointer"
+          >
+            External lightning wallet
+          </Label>
+        </div>
+      </RadioGroup>
+      {!isInternalSwap && (
+        <div className="grid gap-1.5">
+          <Label>Receiving lightning address</Label>
+          <div className="flex gap-2 mb-4">
+            <Input
+              placeholder="satoshi@getalby.com"
+              value={destination}
+              onChange={(e) => setDestination(e.target.value)}
+              required
+            />
+            <Button
+              type="button"
+              variant="outline"
+              className="px-2"
+              onClick={paste}
+            >
+              <ClipboardPasteIcon className="w-4 h-4" />
+            </Button>
+          </div>
+        </div>
+      )}
+      {/* TODO: Review fee for swap ins */}
+      <div className="flex items-center justify-between border-t py-4">
+        <Label>Fee</Label>
+        <p className="text-muted-foreground text-sm">
+          {swapsSettings.albyServiceFee + swapsSettings.boltzServiceFee}% +
+          on-chain fees
+        </p>
+      </div>
+      <LoadingButton loading={loading}>Swap In</LoadingButton>
+    </form>
+  );
+}
+
+function SwapOutForm() {
+  const { toast } = useToast();
+  // TODO: Optimize by setting this from the backend
+  const { data: onchainAddress } = useOnchainAddress();
+  const { data: swapsSettings, mutate } = useSwaps();
+
+  const [swapTo, setSwapTo] = useState("internal");
+  const [balanceThreshold, setBalanceThreshold] = useState("");
+  const [swapAmount, setSwapAmount] = useState("");
+  const [destination, setDestination] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [showAdvanced, setShowAdvanced] = useState(false);
+  const [recurringSwaps, setRecurringSwaps] = useState(false);
+
+  const onSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    try {
+      setLoading(true);
+      await request(`/api/wallet/${recurringSwaps ? "swaps" : "swap"}`, {
```

Comment:
I don't really like this, we are using different endpoints AND passing in different arguments. Can we split it up?